controllers/payment.controllers.js

```diff
@@ -190,73 +260,144 @@ async function userTransactionsHistory(req, res, next) {
+        'You should add an account to hold money first.',
         500,
       );
       return next(error);
     }
 
-    const transfer = await createTransfer({
-      amount,
-      destination: connectedAccount,
-    });
-    if (!transfer) {
+    const amountInDollars = amount * 100;
+    let transfer;
+    try {
+      transfer = await createTransfer({
+        amount: amountInDollars,
+        destination: connectedAccount,
+      });
+    } catch (err) {
       const error = new HttpError(
-        'Could not send funds. Please try again later.',
+        'Could not create transfer. Please try again later.',
         500,
       );
       return next(error);
     }
-    await instantPayout({ amount, destination: connectedAccount });
-    const balance = await changeBalance({ stripeUserId: basketId, amount });
-    if (!balance) {
+
+    try {
+      await instantPayout({ amount: amountInDollars, destination: connectedAccount });
+    } catch (err) {
+      new HttpError(
+        "Sorry, your country doesn't support instant payouts. Stripe will send your funds to your bank account within a few days.",
+        500,
+      );
+    }
+
+    try {
+      await changeBalance({
+        stripeUserId: basket.stripeId,
+        amount: amountInDollars,
+        description: `Payouts from ${basket.name}`,
+      });
+    } catch (err) {
       const error = new HttpError(
         'Could not send funds. Please try again later.',
         500,
       );
       return next(error);
     }
-    const paymentMethod = await getConnectedCard('connectedAccount');
-    const transaction = await createTransaction({
-      basketId: basket._id,
-      userId: id,
-      stripeId: transfer,
-      amount: amount,
-      comment: `Payouts from ${basket_name}`,
-      paymentMethod: paymentMethod,
-    });
-    res.status(200).json(transaction);
+
+    let paymentMethod;
+    try {
+      paymentMethod = await getConnectedCard({accountId: connectedAccount});
+    } catch (err) {
+      const error = new HttpError(
+        'Could find your card to send funds. Please try again later.',
+        500,
+      );
+      return next(error);
+    }
+
+    try{
+        basket.value -= amount;
+        await basket.save();
+    }catch(err){
+        const error = new HttpError(
+          'Could find your card to send funds. Please try again later.',
+          500,
+        );
+        return next(error);
+    }
+
+    let transaction;
+    try{
+        transaction = await createTransaction({
+          basketId: basket._id,
+          userId: req.user._id,
+          stripeId: transfer,
+          amount: amount,
+          comment: `Payouts from ${basket.name}`,
+          card: paymentMethod.last4,
+        });
+    }catch(err){
+        const error = new HttpError(
+          'Could not create transaction. Please try again later.',
+          500,
+        );
+        return next(error);
+    }
+    res.status(200).json(transaction.status);
```

Comment:
```suggestion
    try{
        сщтіеtransaction = await createTransaction({
          basketId: basket._id,
          userId: req.user._id,
          stripeId: transfer,
          amount: amount,
          comment: `Payouts from ${basket.name}`,
          card: paymentMethod.last4,
        });
        
        res.status(200).json(transaction.status);

    }catch(err){
        const error = new HttpError(
          'Could not create transaction. Please try again later.',
          500,
        );
        return next(error);
    }
```

Reply:
move res.status... into try block, then we can use cont transaction instead of let

Reply:
done